packages/auth/src/app.ts

```diff
@@ -153,10 +154,35 @@ export class App {
       resolvers
     })
 
+    // Setup Armor
+    const armor = new ApolloArmor({
+      blockFieldSuggestion: {
+        enabled: true
+      },
+      maxDepth: {
+        enabled: true,
+        n: 10,
+        ignoreIntrospection: false
+      },
+      costLimit: {
+        enabled: true,
+        maxCost: 5000,
+        objectCost: 2,
+        scalarCost: 1,
+        depthCostFactor: 1.5,
+        ignoreIntrospection: true
+      }
+    })
+    const protection = armor.protect()
+
     // Setup Apollo
     const apolloServer = new ApolloServer({
       schema: schemaWithResolvers,
-      plugins: [ApolloServerPluginDrainHttpServer({ httpServer })],
+      ...protection,
+      plugins: [
+        ...protection.plugins,
+        ApolloServerPluginDrainHttpServer({ httpServer })
+      ],
```

Comment:
I realize this is merged but wanted to comment here. We can add new issues for this. 

The addition of `...protection` breaks introspection. It's working on main (before this was merged) but not here. Commenting out `...protection` fixes it.

And I know this is how the docs show it, but destructuring protection is probably not the best way to add it IMO because it has a plugins array. If we were to change the order of `...protection` and `plugins` on this object then the plugin array we are forming ourslves would get overridden:

```ts
const apolloServer = new ApolloServer({
      schema: schemaWithResolvers,
      plugins: [
        ...protection.plugins,
        ApolloServerPluginDrainHttpServer({ httpServer })
      ],
      ...protection, // overrides the above plugins
      introspection: this.config.env !== 'production'
    })
```

Again, it's techniaclly fine as-is but it's not obvious how close it is to not fine. I think something like this would be safer:

```ts
    const { plugins: protectionPlugins, ...protectionRest } = protection

    this.apolloServer = new ApolloServer({
      schema: schemaWithMiddleware,
      ...protectionRest,
      plugins: [
        ...protectionPlugins,
        ApolloServerPluginDrainHttpServer({ httpServer })
      ],
      introspection: this.config.env !== 'production'
    })
```

Reply:
it's the `maxDepth.ignoreIntrospection` breaking it. Although I see that's intentionally set that way to resolve an issue with max depth being ignored. Maybe it can also be controlled by NODE_ENV (not disabled in not prod)?